### ISSUE TITLE
DialogModal misaligns child RadioButton text

### ISSUE THREAD

**Issue by morgan-sam** [curology/radiance-ui#369]
### Issue

When a `<RadioButton />` component is placed inside a `<DialogModal />` the text will be misaligned. Example:

```
import React from "react";
import { DialogModal, RadioButton } from "radiance-ui";

function App() {
  return (
    <DialogModal>
      <RadioButton type="primary" checked>
        This is a primary radio button
      </RadioButton>
    </DialogModal>
  );
}

export default App;
```

Will result in:

![misalignedRadioButton](https://user-images.githubusercontent.com/57941781/93022878-532c1280-f5e3-11ea-89b1-608ccb162996.png)

### Cause

This is caused by the margin-bottom styles applied to all `p` & `p:last-of-type` elements inside `<DialogModal />`; including nested `p` elements inside child components.  [This is the responsible styling](https://github.com/curology/radiance-ui/blob/a6a8e35466570e4dcb612a788c827e6c73579dd1/src/shared-components/immersiveModal/style.ts#L93).

### Workaround

This can be temporarily fixed by using emotion to inject a css style to override the margin styling: 

```
import React from "react";
import { DialogModal, RadioButton } from "radiance-ui";
import { css } from "@emotion/core";

function App() {
  return (
    <DialogModal>
      <RadioButton
        type="primary"
        checked
        css={css`
          & > p:last-of-type {
            margin-bottom: 0;
          }
        `}
      >
        This is a primary radio button
      </RadioButton>
    </DialogModal>
  );
}

export default App;
```

Will result in:

![alignedRadioButton](https://user-images.githubusercontent.com/57941781/93023089-7c996e00-f5e4-11ea-98c0-3168473104db.png)

**Comment by michaeljaltamirano** [comment]
@morgan-sam Thank you for pointing this out! It looks like our usage (outside of `radiance-ui`) has a regular pattern of nesting paragraph tags like so:

```tsx
<DialogModal title="We changed things up to give you more control">
  <p>
    Example Text
  </p>
  {...rest}
</DialogModal>
```

Based on your example code, I think an appropriate course of action is to codify the convention of a paragraph tag child as a prop:

```tsx
<DialogModal title="We changed things up to give you more control" message="Example Text">
  {...rest}
</DialogModal>
```

We can then wrap the message in a special styled component and use that as the selector instead of p tags, generically:

```ts
export const Message = styled.p``;

export const ModalBody = styled.div`
  ${Message} {
    margin-bottom: ${SPACER.large};
    &:last-of-type {
      margin-bottom: ${SPACER.xlarge};
    }
  }
`;

That way these styles won't bleed over to other components, unnecessarily.

I'll update this issue once the PR is ready, for tracking purposes. Again, thank you for the issue!
```

**Comment by zcaceres** [comment]
Encountered this issue while creating a custom Modal. IMO it seems strange that the "Container" (ModalBody) is styling its child elements. Perhaps there is a good reason for this that I don't know yet.

But leaving the container responsible for its own styles and behavior and letting the engineer compose spacing and styling for the child elements strikes me as a more standard pattern for UI libraries.

**Comment by michaeljaltamirano** [comment]
This will be fixed in `v18.0.0`, PR: https://github.com/curology/radiance-ui/pull/833

Closing. Thanks again for pointing this out @morgan-sam 🙏
